consumer/converter.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package consumer contains interfaces that receive and process consumerdata.
+package consumer
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/translator/internaldata"
+)
+
+// NewBackwardTraceConverter creates new backwardTraceConverter that takes TraceConsumer and implements ConsumeTraceV2
+func NewBackwardTraceConverter(tc TraceConsumer) TraceConsumerV2 {
+	return &backwardTraceConverter{tc}
+}
+
+// backwardTraceConverter is a internal->oc translation shim that takes TraceConsumer and implements ConsumeTraceV2 interface
+type backwardTraceConverter struct {
```

Comment:
Can we use a more specific name? "backward" is very context dependent name. How about "internalToOcTraceConverter"?

Reply:
Makes sense, changed.